### PR TITLE
bazel: update gtest

### PR DESCRIPTION
**Pull request by keith**
This is required for working with newer versions of bazel

Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

### CODE REVIEW

**Comment by RyanTheOptimist** [comment]
Any idea why CI is red?

**Comment by keith** [comment]
sorry i thought i was past it, will investigate, there's some weirdness around how gtest uses a branch of re2

**Comment by keith** [comment]
pinning back, hoping this works, filed https://github.com/google/re2/issues/388

**Comment by RyanTheOptimist** [comment]
Looks like this is still having trouble with CI.

/wait

**Comment by junyer** [comment]
[`com_googlesource_code_re2`](https://github.com/envoyproxy/envoy/blob/4632a68618322c482e3b750adf0805ed9c01460a/bazel/repository_locations.bzl#L1042-L1053) needs to point to https://github.com/google/re2/commit/6543c379df6054c3926bb5b6bdc0869d1e6b099b now, @keith?

**Comment by keith** [comment]
I kept the older gtest sha that still doesn't include re2 for now and I figured I would update it again + re2 later once this was green, thanks for pushing that fix!

**Comment by RyanTheOptimist** [comment]
Are these failures related, or just flakes?

**Comment by keith** [comment]
They're related. Something changed semantically that I need to debug

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -730,13 +730,11 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
-        # Pick up fix for MOCK_METHOD compilation with clang-cl for Windows (resolved after 1.10.0)
-        # see https://github.com/google/googletest/issues/2490
-        version = "a4ab0abb93620ce26efad9de9296b73b16e88588",
-        sha256 = "7897bfaa5ad39a479177cfb5c3ce010184dbaee22a7c3727b212282871918751",
+        version = "0498660ea575bfeb4b3b0879fa6aa6904d1df373",
+        sha256 = "63222f9431a512cf8c82fc9953876bf3c9c23c7074096f029a3213501d210b2a",
         strip_prefix = "googletest-{version}",
         urls = ["https://github.com/google/googletest/archive/{version}.tar.gz"],
-        release_date = "2020-09-10",
+        release_date = "2022-04-26",
```
Comment (moderation) [inline bazel/repository_locations.bzl:737]: Can this dependency switch to the `v1.12.1` release? That release contains this commit. https://github.com/google/googletest/releases/tag/release-1.12.1

**Review by moderation — CHANGES_REQUESTED** [review]

**Pull request closed without merging**